plugins/modules/job_invocation.py

```diff
@@ -215,7 +218,12 @@ def main():
                 module.foreman_params['bookmark']),
                 failsafe=False,
             ))
-        module.run()
+        module.auto_lookup_entities()
+        entity = module.lookup_entity('entity')
+        if entity is None:
+            module.ensure_entity('job_invocations', module.foreman_params, entity, state='present')
+        else:
+            module.fail_json(msg="You cannot update an existing job_invocation")
```

Comment:
Can this ever happen? I assumed, this module would just trigger the same command a second time (which when used as a handler would be the desired outcome, right?).
It would look something like:

```suggestion
        module.auto_lookup_entities()
        module.ensure_entity('job_invocations', module.foreman_params, None, state='present')
```

Reply:
Yepp, that's I also wanted to suggest.

I suggested the if/else in one of my previous comments, but given we can't (and don't) search for an entity anymore, this is more correct

Reply:
And all the `record_before/after[full]` will be handled magically this way, right?

Reply:
`ensure_entity` handles all the before/after recording, yes.

but the reporting is only part of `ForemanEntityAnsibleModule`, :thinking:

Reply:
but we can fix that in a separate PR I think

Reply:
opened https://github.com/theforeman/foreman-ansible-modules/issues/996 for tracking it